doc/source/whatsnew/v0.21.0.txt

```diff
@@ -79,6 +79,62 @@ Other Enhancements
 - :func:`date_range` now accepts 'YS' in addition to 'AS' as an alias for start of year (:issue:`9313`)
 - :func:`date_range` now accepts 'Y' in addition to 'A' as an alias for end of year (:issue:`9313`)
 
+.. _whatsnew_0210.enhancements.read_csv:
+
+``read_csv`` use `python-requests` (if installed) to support basic auth and much more
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If `python-requests` library is installed try to use it first. If not, continue using urllib
```

Comment:
> I want to be prepared and find issues early. How should i force pandas to use requests, even if I am not using http auth? How else do you expect to gain confidence in requests and detect scenarios of inconsistencies?

That's fair, but you as the user should have to go out of your way to use it until we start enabling as the default.  Notice I didn't prevent you at this point from passing in a non-empty dictionary, even if it contains parameters that are ultimately default parameters for `requests`.